### DISCUSSION TITLE
What's the best way to transition from Java EE applications with multiple shared libraries to Quarkus?

### DISCUSSION THREAD

**Discussion by hfluz** [quarkusio/quarkus#37589] [Q&A]
We currently develop on Java EE 8 / JSF + Mybatis + Hibernate / Wildfly 26 and we have more than 30 web projects and around 8 database projects.

A database project contain the models, DAOs and business/validation rules and may be used by several Web projects.

Currently we can open all the projects in which we are going to work with at the moment on IntelliJ IDEA. We have hot deploy/reload available for any of their Java classes and xhtml files (Compile and reload file / update resources). That's a great boost for our productivity and something we can't find in any other IDE.

We've been considering to migrate to Quarkus and I've already developed a pet project that's working the same framework stack we use on Wildfly, but I'm still not sure the best way to transition. As far as I learned, Quarkus live reload will work only on the main project. If I change something on a database projects or other dependencies, I have to rebuild it with maven and build the Web project again with Quarkus.

I like getting a live/hot reload that is IDE agnostic, but losing this feature on the other dependencies would feel like a downgrade in our development experience. Could someone provide some guidance on the better way to transition to Quarkus, considering the way our projects are structured? Would we have to convert all the database projects to web services and run several Quarkus instances simultaneously in local environment? Or am I missing some Quarkus feature that would help us?

**Comment by varijkapil13** [comment]
This maybe a bit old,  but I found this video that might be of help to you.

https://www.youtube.com/watch?v=nN1iYPFU_1k

**Comment by mapuci** [comment]
We faced the exact same issue, fix/workaround is described here https://github.com/quarkusio/quarkus/issues/33725
See `aggregator pom` concept... I aggree its not perfect and that it should be more documented, but that is the only way I know how to make quarkus compile projects from multiple git repos at the same time.
